synapse/lib/crypto/coin.py

```diff
@@ -76,7 +76,10 @@ def ether_eip55(body: str):
 
 def eth_check(match: regex.Match):
     text = match.groupdict().get('valu')
-    prefix, body = text.split('x')  # type: str, str
+    if text.find('0x') != -1:
```

Comment:
This find/split will be inefficient. Would be better to `text.startswith('0x') or text.startswith('0X'): text = text[2:]`